sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -0,0 +1,476 @@
+ * @return  NULL, if no space is left.
+ */
+_nib_t *_nib_alloc(const ipv6_addr_t *addr, unsigned iface);
+
+/**
+ * @brief   Clears out a NIB entry (on-link version)
+ *
+ * @param[in] nib  An entry
+ *
+ * @return  true, if entry was cleared
+ * @return  false, if entry was not cleared
+ */
+static inline bool _nib_clear(_nib_t *nib)
+{
+    if (nib->mode == _EMPTY) {
+        memset(nib, 0, sizeof(_nib_t));
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief   Iterates over on-link entries
+ *
+ * @param[in] last  Last entry (NULL to start)
+ *
+ * @return  entry after @p last
+ */
+_nib_t *_nib_iter(const _nib_t *last);
+
+/**
+ * @brief   Gets a node by IPv6 address and interface
+ *
+ * @pre     `(addr != NULL)`
+ *
+ * @param[in] addr  The address of a node. May not be NULL.
+ * @param[in] iface The interface to the node. May be 0 for any interface
+ *
+ * @return  The NIB entry for node with @p addr and @p iface on success.
+ * @return  NULL, if there is no such entry.
+ */
+_nib_t *_nib_get(const ipv6_addr_t *addr, unsigned iface);
+
+/**
+ * @brief   Adds a node to the neighbor cache
+ *
+ * @pre     `(addr != NULL)`
+ * @pre     `(cstate != GNRC_IPV6_NIB_NC_INFO_NUD_STATE_DELAY)`
+ * @pre     `(cstate != GNRC_IPV6_NIB_NC_INFO_NUD_STATE_PROBE)`
+ * @pre     `(cstate != GNRC_IPV6_NIB_NC_INFO_NUD_STATE_REACHABLE)`
+ *
+ * @param[in] addr      The address of a node. May not be NULL.
+ * @param[in] iface     The interface to the node.
+ * @param[in] cstate    Creation state. State of the entry *if* the entry is
+ *                      newly created.
+ * @return  The NIB entry for the new neighbor cache on success.
+ * @return  NULL, if there is no space left.
+ */
+_nib_t *_nib_nc_add(const ipv6_addr_t *addr, unsigned iface, uint16_t cstate);
+
+/**
+ * @brief   Removes a node from the neighbor cache
+ *
+ * @param[in] nib   A node.
+ */
+void _nib_nc_remove(_nib_t *nib);
+
+/**
+ * @brief   Sets a NUD-managed neighbor cache entry to reachable and sets the
+ *          respective event in @ref _nib_evtimer "event timer"
+ *          (@ref GNRC_IPV6_NIB_MSG_NUD_SET_STALE)
+ *
+ * @param[in] nib   A node.
+ */
+void _nib_nc_set_reachable(_nib_t *nib);
+
+/**
+ * @brief   Creates or gets an existing default router list entry by address
+ *
+ * @pre     `(router_addr != NULL)`
+ *
+ * @param[in] addr  An IPv6 address. May not be NULL.
+ *                  *May also be a global address!*
+ * @param[in] iface The interface to the router.
+ *
+ * This creates or gets an existing default router entry by address. Older
+ * on-link entries might be cycled out by this for _nib_dr_t::next_hop in a
+ * CLOCK-wise manner, if they are non-persistent cache entries.
+ *
+ * @return  A new or existing default router entry with _nib_t::ipv6 of
+ *          _nib_dr_t::next_hop set to @p router_addr.
+ * @return  NULL, if no space is left.
+ */
+_nib_dr_t *_nib_drl_add(const ipv6_addr_t *router_addr, unsigned iface);
```

Comment:
It's a abstraction layer thing. The 'view-layer' add functions are named `_add`, the data-layer functions are named `_alloc` (compare `_nib_nc_add` vs `_nib_alloc`). `_nib_drl_add` sits in both view- and data layer, since the object type default router exist in both. I could write a wrapper for the view-layer around the data-layer function, but that would just be a call of that. That's why I called the functions on these objects all like the functions on the data layer.